src/modal/modal.js

```diff
@@ -414,12 +414,13 @@ angular.module('ui.bootstrap.modal', ['ui.bootstrap.stackedMap'])
           backdrop: modal.backdrop,
           keyboard: modal.keyboard,
           openedClass: modal.openedClass,
-          windowTopClass: modal.windowTopClass
+          windowTopClass: modal.windowTopClass,
+          container: modal.container
         });
 
         openedClasses.put(modalBodyClass, modalInstance);
 
-        var body = $document.find('body').eq(0),
+        var container = $document.find(modal.container).eq(0),
```

Comment:
I still think we need to only grab the first one.  If I'm reading [this](http://api.jquery.com/append) right, if someone passed in multiple objects, the modal would get appended to each of them.